tonic/src/transport/server/mod.rs

```diff
@@ -827,7 +862,8 @@ fn serve_connection<B, IO, S, E>(
 
             let mut conn = pin!(builder.serve_connection(hyper_io, hyper_svc));
 
-            let mut sleep = pin!(sleep_or_pending(max_connection_age));
+            let mut graceful_sleep = pin!(sleep_or_pending(max_connection_age));
+            let mut forceful_sleep = pin!(sleep_or_pending(None));
```

Comment:
Am I missing something but we should be able to combine these two and their logic together into a singular future that is `max age wait -> force ful wait -> exit`? From this we can write a simple unit test using tokio `test-utils` with their test timer?